python/ray/tune/suggest/variant_generator.py

```diff
@@ -253,10 +253,18 @@ def get_preset_variants(spec: Dict,
                 f"this path to the `config` variable passed to `tune.run()`."
             ) from exc
 
-        if domain and not domain.is_valid(val):
-            logger.warning(
-                f"Pre-set value `{val}` is not within valid values of "
-                f"parameter `{'/'.join(path)}`: {domain.domain_str}")
+        if domain:
+            if isinstance(domain, Domain):
+                if not domain.is_valid(val):
+                    logger.warning(
+                        f"Pre-set value `{val}` is not within valid values of "
+                        f"parameter `{'/'.join(path)}`: {domain.domain_str}")
+            else:
+                # domain is actually a fixed value
+                if domain != val:
+                    logger.warning(
+                        f"Pre-set value `{val}` is not equal to the value of "
+                        f"parameter `{'/'.join(path)}`: {domain}")
```

Comment:
Can this ever happen?

Reply:
Yes, like this
```python
from ray import tune
from ray.tune.suggest.basic_variant import BasicVariantGenerator


tune.run(
    lambda config: config["a"] + config["b"],
    config={
        "a": 1,
        "b": tune.randint(0, 3)
    },
    search_alg=BasicVariantGenerator(points_to_evaluate=[
        {"a": 2, "b": 2},
    ]),
    num_samples=1)
```
The values of `a` aren't consistent, but I see no exception.

Further more, I agree with throwing an exception instead of throwing a warning

Reply:
Ah I see - in that case, let's leave it as is (don't throw an exception, just a warning).

The use case I have in mind is that someone might want to try out specific values for `a` but keep it constant in random exploration.

Reply:
So basically you'll just need to add a unit test then - otherwise looks good!

Reply:
So in which file the unit test should be added?